modules/libcom/src/iocsh/libComRegister.c

```diff
@@ -396,13 +398,39 @@ static void installLastResortEventProviderCallFunc(const iocshArgBuf *args)
 
 static iocshVarDef asCheckClientIPDef[] = { { "asCheckClientIP", iocshArgInt, 0 }, { NULL, iocshArgInt, NULL } };
 
+// Register the PWD environment variable when the cd IOC shell function is 
+// registered. This variable contains the current directory path.
+static void registerPwd() {
+    // On Linux, the IOC already boots with the PWD env variable initialized.
+    // We check if it exists thinking about other OSs.
+    char* currentPwd = getenv("PWD");
+    if (currentPwd == NULL) {
```

Comment:
I think this test shoulf be removed.  `$PWD` should always be in sync with `getcwd()`, so there is no harm in enforcing this.  As written this code path will almost never run on Linux, which is a recipe for rare bugs.

Reply:
Ok, I removed the test.